sdk/cosmos/azure-cosmos/test/test_container_properties_cache.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# The MIT License (MIT)
+# Copyright (c) Microsoft Corporation. All rights reserved.
+
+"""End-to-end test.
+"""
+
+import unittest
+import uuid
+
+import pytest
+
+import azure.cosmos.cosmos_client as cosmos_client
+import azure.cosmos.exceptions as exceptions
+import test_config
+from azure.cosmos.partition_key import PartitionKey
+
+
+@pytest.mark.cosmosEmulator
+class TestContainerPropertiesCache(unittest.TestCase):
+    """Python CRUD Tests.
+        """
+
+    configs = test_config.TestConfig
+    host = configs.host
+    masterKey = configs.masterKey
+    connectionPolicy = configs.connectionPolicy
+    last_headers = []
+    client: cosmos_client.CosmosClient = None
+
+    def __AssertHTTPFailureWithStatus(self, status_code, func, *args, **kwargs):
+        """Assert HTTP failure with status.
+
+                :Parameters:
+                    - `status_code`: int
+                    - `func`: function
+                """
+        try:
+            func(*args, **kwargs)
+            self.assertFalse(True, 'function should fail.')
+        except exceptions.CosmosHttpResponseError as inst:
+            self.assertEqual(inst.status_code, status_code)
+
+    @classmethod
+    def setUpClass(cls):
+        if (cls.masterKey == '[YOUR_KEY_HERE]' or
```

Comment:
for these tests, are you storing **masterKey** and **host** as an environment variable? Is this test ran gated?